### ISSUE TITLE
[Accessibility] Microsite - missing link titles

### ISSUE THREAD

**Issue by emmaindal** [backstage/backstage#21883]
### 📜 Description

Missing on Home page

Link title is crucial, either as visible text or an accessible label to satisfy [2.4.4 Link Purpose](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html)

### 👍 Expected behavior

[Links to have discernible text](https://dequeuniversity.com/rules/axe/4.8/link-name)

### 👎 Actual Behavior with Screenshots

Example: 
<img width="586" alt="Screenshot 2023-12-15 at 13 22 44" src="https://github.com/backstage/backstage/assets/25153114/a5519d5a-5d67-46d9-a3ea-b93c26402d06">

### 👟 Reproduction steps

1. Go to https://backstage.io/
2. [Run the Lighthouse browser extension ](https://chromewebstore.google.com/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?pli=1)

Alternatively:
1. Run the microsite locally
2. [Run the Lighthouse CLI locally when developing new features](https://backstage.io/docs/accessibility/#run-the-lighthouse-cli-locally-when-developing-new-features)

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None
